.github/workflows/CI.yml

```diff
@@ -0,0 +1,65 @@
+name: full_tests
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
+  schedule:
+    # nightly tests, 2 am
+    - cron: "0 2 * * *"
+
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-10.15, macos-11]
+        python-version: ["3.8", "3.9",]
```

Comment:
can probably stay slim and just pick a version of python here

Reply:
:+1: